ledger/participant-state/kvutils/src/test/lib/scala/com/daml/ledger/participant/state/kvutils/export/LedgerDataExportSpecBase.scala

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.export
+
+import java.io.{InputStream, OutputStream, PipedInputStream, PipedOutputStream}
+import java.time.Instant
+
+import com.daml.ledger.participant.state.v1
+import com.google.protobuf.ByteString
+import org.scalatest.{Matchers, WordSpec}
+
+abstract class LedgerDataExportSpecBase(name: String) extends WordSpec with Matchers {
+  protected def newExporter(outputStream: OutputStream): LedgerDataExporter
+
+  protected def newImporter(inputStream: InputStream): LedgerDataImporter
+
+  name should {
+    "serialize a submission to something deserializable" in {
+      val submissionInfo = SubmissionInfo(
+        participantId = v1.ParticipantId.assertFromString("id"),
+        correlationId = "parent",
+        submissionEnvelope = ByteString.copyFromUtf8("an envelope"),
+        recordTimeInstant = Instant.ofEpochSecond(123456, 123456789),
+      )
+
+      val inputStream = new PipedInputStream()
+      val outputStream = new PipedOutputStream(inputStream)
+      val exporter = newExporter(outputStream)
+
+      val submission = exporter.addSubmission(submissionInfo)
+      val writeSetA1 = submission.addChild()
+      writeSetA1 ++= Seq(keyValuePairOf("a", "b"), keyValuePairOf("c", "d"))
+      val writeSetA2 = submission.addChild()
+      writeSetA2 ++= Seq(keyValuePairOf("e", "f"), keyValuePairOf("g", "h"))
+
+      submission.finish()
+
+      val importer = newImporter(inputStream)
+
+      val (actualSubmissionInfo, actualWriteSet) = importer.read().head
+      actualSubmissionInfo should be(submissionInfo)
+      actualWriteSet should be(
+        Seq(
+          keyValuePairOf("a", "b"),
+          keyValuePairOf("c", "d"),
+          keyValuePairOf("e", "f"),
+          keyValuePairOf("g", "h"),
+        ))
+    }
+  }
```

Comment:
Test coverage seems a bit on the thin side but I think that's out of scope for this PR.

Reply:
I agree, I'd like to add more.

Worth noting we also run an integration check by using the test tool and in the integrity checker, which should catch most issues at a high level. So I'm not too worried.